app.rb

```diff
@@ -60,22 +62,52 @@ def create_book
   end
 
   def create_rental
-    puts 'Select book number'
-    @books.each_with_index do |book, index|
-      puts "#{index}. Title: #{book.title}, Author: #{book.author}"
-    end
-    book_index = gets.chomp.to_i
-    puts 'Select from the following list'
-    @people.each_with_index do |person, index|
-      type = person.instance_of?(Student) ? 'Student' : 'Teacher'
-      puts "#{index}. [#{type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
-    end
-
-    person_index = gets.chomp.to_i
+    person_index = select_person
+    book_index = select_book
     print 'Date: '
     date = gets.chomp
     @rentals.push(Rental.new(date, @books[book_index], @people[person_index]))
     puts 'Rentel successfully created'
+    save_data
+  end
+
+  def list_rentals(person_id)
+    rentals_files = ReadFile.new('rentals.json').read || []
+    person_rentals = rentals_files.select { |rental| rental['person']['id'] == person_id }
+
+    if person_rentals.empty?
+      puts 'No available rentals for person with that ID.'
+    else
+      person_rentals.each do |rental|
+        date = rental['date']
+        book_title = rental['book']['title']
+        book_author = rental['book']['author']
+        person_type = rental['person']['type']
+        person_id = rental['person']['id']
+        person_age = rental['person']['age']
+        person_name = rental['person']['name']
+
+        puts "Date: #{date}"
+        puts "Book Title: #{book_title}"
+        puts "Book Author: #{book_author}"
+        puts "Person Type: #{person_type}"
+        puts "person ID: #{person_id}"
+        puts "Person Age: #{person_age}"
+        puts "Person Name: #{person_name}"
+      end
+    end
+  end
+
+  def select_book
+    puts 'Select book ffrom the number: '
+    list_people
```

Comment:
- [ Required ] I think in this `select_book` method you should list all the books available and not the _people_ this method has been called in the `create_rental` method which confuses a user who is trying to rent a book since the `people` are listed twice and no `book` is listed. Please see the screenshot below and rectify this.

## Error in listing books to rent
![Screenshot from 2023-09-15 00-48-04](https://github.com/H-Ramalan/school-library/assets/16392046/a86c62b2-9766-4481-9fa8-37748e402a27)